packages/php-wasm/universal/src/lib/rethrow-file-system-error.ts

```diff
@@ -6,10 +6,17 @@
  * @see https://github.com/emscripten-core/emscripten/blob/38eedc630f17094b3202fd48ac0c2c585dbea31e/system/include/wasi/api.h#L336
  */
 
-export interface ErrnoError extends Error {
+export class ErrnoError extends Error {
+	constructor(errno: number, message?: string, options?: { cause?: any }) {
+		super(message);
+		this.name = 'ErrnoError';
+		this.errno = errno;
+		this.message = message ?? '';
+		this.cause = options?.cause;
+	}
+
 	node?: any;
 	errno: number;
-	message: string;
```

Comment:
Why are we assigning to `this.message` above if we are removing its declaration? Can we rely upon the `Error` super constructor to set `message` correctly?

Reply:
In addition, should we be passing options with `cause` to the super constructor and skipping assignment here?